SingletonType bindings not working since 4.2.2

Since version `4.2.2` and upwards, this binding results in an exception:

```scala
// A sealed domain
sealed trait Root
final case object Foo extends Root
final case object Bar extends Root
final case object Baz extends Root

// A case class using said domain
case class MyContainerClass[T <: Root](inner: T)

// In a ScalaModule:
bind[MyContainerClass[Foo.type]].toProvider[FooProvider]
bind[MyContainerClass[Bar.type]].toProvider[BarProvider]
bind[MyContainerClass[Baz.type]].toProvider[BazProvider]
```

This results in the following exception:

`[Error] Caused by: java.lang.UnsupportedOperationException: Could not convert scalaType Foo.type to a javaType: scala.reflect.internal.Types$UniqueSingleType at net.codingwell.scalaguice.TypeConversions$.scalaTypeToJavaType(TypeConversions.scala:78)`

With `4.2.1`, this works without a problem. Am I doing something wrong, or is this type of binding simply not supported (anymore), @tsuckow? Thanks in advance for the assistance. :+1:

--- a/src/main/scala/net/codingwell/scalaguice/TypeConversions.scala
+++ b/src/main/scala/net/codingwell/scalaguice/TypeConversions.scala
@@ -6,8 +6,7 @@ import com.google.inject.internal.MoreTypes.WildcardTypeImpl
 import com.google.inject.util.Types._
 
 import scala.reflect.runtime.universe
-import scala.reflect.runtime.universe._
-import scala.reflect.runtime.universe.{Type => ScalaType, WildcardType => ScalaWildCardType}
+import scala.reflect.runtime.universe.{Type => ScalaType, _}
 
 /**
   * Copyright (C) 22/04/2018 - REstore NV
@@ -74,6 +73,14 @@ private [scalaguice] object TypeConversions {
         val mappedUpperBounds = upperBounds.map(bound => scalaTypeToJavaType(bound, mirror)).toArray
         val mappedLowerBounds = lowerBounds.map(bound => scalaTypeToJavaType(bound, mirror)).toArray
         new WildcardTypeImpl(mappedUpperBounds, mappedLowerBounds)
+      }
+      case SingleType(_, symbol) if symbol.isModule => {
+        val rm = universe.runtimeMirror(getClass.getClassLoader)
+        val moduleReflection = rm.reflectModule(symbol.asModule)
+        val instanceMirror = mirror.reflect(moduleReflection.instance)
+        val classMirror = rm.reflectClass(instanceMirror.symbol)
+
+        scalaTypeToJavaType(classMirror.symbol.toType, mirror)
       }
       case _ => throw new UnsupportedOperationException(s"Could not convert scalaType $scalaType to a javaType: " + scalaType.dealias.getClass.getName)
     }

